### ISSUE TITLE
Documentation for upgrading from phpWebSite

### ISSUE THREAD

**Issue by obones** [AppStateESS/canopy#1]
I'm using a quite old version of phpWebSite (from 2010) on some personal projects and I would like to upgrade them to Canopy.
Are there any specific instructions on how to do this? I could not find anything in the repository, but maybe I did not look in the appropriate place.

**Comment by Stardog** [comment]
It has been so long someone outside our university asked about the software :)

First off, grab the latest version of phpwebsite and update everything. Assume modules that were previously included with phpwebsite that are now individual repositories are Canopy only. 

You will need php 7.0 or higher. Once you untar, you should be able just point Canopy at the old database and run the updates in Boost.

I will warn you that some older modules have deprecated and simply will not work. If you are using a  third party module, most likely it will give you problems.

Old themes will not work. Look at  bootstrap4-default that comes with Canopy for hints on how to update your theme. 

We are currently updating as we go here. I recommend making a new hub and move each branch over individually. Go through the branch and update the old Bootstrap and Font Awesome icons in your PageSmith content.

Make sure to get the releases of newer modules. They include the npm builds.

If you have other questions, you can ask them here.

**Comment by obones** [comment]
Thanks, it's also been a long time since I had to upgrade anything on those projects.
I'll work steps by steps and ask here if I'm hit by any roadblocks.

**Comment by obones** [comment]
I just downloaded the latest GIT content, placed in on my shared hosting and connected to it with my browser.
It told me that I had no config file, so I clicked on the setup link but was given the following error messages:

Warning: require_once(C:\xampp-php7\htdocs\canopy-master/vendor/autoload.php): failed to open stream: No such file or directory in C:\xampp-php7\htdocs\canopy-master\Autoloaders\composer.php on line 29

Fatal error: require_once(): Failed opening required 'C:\xampp-php7\htdocs\canopy-master/vendor/autoload.php' (include_path='.:C:\xampp-php7\htdocs\canopy-master/lib/pear/') in C:\xampp-php7\htdocs\canopy-master\Autoloaders\composer.php on line 29

Having a look at the README.md file, there is a section that says this:

Before using the web installer you will have to install composer and run composer install from the project root directory

But I don't have any shell access to my hosting, it's only a shared hosting provider.
However, I do have a few Windows or Linux machines here at work, so I could call some shell commands on a local copy. Would this be enough to generate the appropriate files? Will it work once I have uploaded the result to my shared hosting environment?

**Comment by eberhardtm** [comment]
Yes, that "should" work as long as your environment on the linux box is similar to your hosting provider. Its fairly generic stuff that is installed with composer so I can't really see a reason why it wouldn't work. Then copy the vendor directory.

**Comment by Stardog** [comment]
Make sure to install composer. Then run the command 'composer install'
